jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/WarningsNgPublisher.java

```diff
@@ -0,0 +1,130 @@
+        FilePath workspace = context.get(FilePath.class);
+        Run run = context.get(Run.class);
+        Launcher launcher = context.get(Launcher.class);
+
+        List<Element> spotbugsEvents = XmlUtils.getExecutionEventsByPlugin(mavenSpyLogsElt, "com.github.spotbugs", "spotbugs-maven-plugin", "spotbugs", "MojoSucceeded", "MojoFailed");
+
+        if (spotbugsEvents.isEmpty()) {
+            LOGGER.log(Level.FINE, "No com.github.spotbugs:spotbugs-maven-plugin:spotbugs execution found");
+            return;
+        }
+
+        List<SpotBugsReportDetails> spotBugsReportDetails = new ArrayList<>();
+        for (Element spotBugsTestEvent : spotbugsEvents) {
+            String spotBugsEventType = spotBugsTestEvent.getAttribute("type");
+            if (!spotBugsEventType.equals("MojoSucceeded") && !spotBugsEventType.equals("MojoFailed")) {
+                continue;
+            }
+
+            Element pluginElt = XmlUtils.getUniqueChildElement(spotBugsTestEvent, "plugin");
+            Element xmlOutputDirectoryElt = XmlUtils.getUniqueChildElementOrNull(pluginElt, "xmlOutputDirectory");
+            Element projectElt = XmlUtils.getUniqueChildElement(spotBugsTestEvent, "project");
+            MavenArtifact mavenArtifact = XmlUtils.newMavenArtifact(projectElt);
+            MavenSpyLogProcessor.PluginInvocation pluginInvocation = XmlUtils.newPluginInvocation(pluginElt);
+
+            if (xmlOutputDirectoryElt == null) {
+                listener.getLogger().println("[withMaven] No <xmlOutputDirectoryElt> element found for <plugin> in " + XmlUtils.toString(spotBugsTestEvent));
+                continue;
+            }
+            String xmlOutputDirectory = xmlOutputDirectoryElt.getTextContent().trim();
+            if (xmlOutputDirectory.contains("${project.build.directory}")) {
+                String projectBuildDirectory = XmlUtils.getProjectBuildDirectory(projectElt);
+                if (projectBuildDirectory == null || projectBuildDirectory.isEmpty()) {
+                    listener.getLogger().println("[withMaven] '${project.build.directory}' found for <project> in " + XmlUtils.toString(spotBugsTestEvent));
+                    continue;
+                }
+
+                xmlOutputDirectory = xmlOutputDirectory.replace("${project.build.directory}", projectBuildDirectory);
+
+            } else if (xmlOutputDirectory.contains("${basedir}")) {
+                String baseDir = projectElt.getAttribute("baseDir");
+                if (baseDir.isEmpty()) {
+                    listener.getLogger().println("[withMaven] '${basedir}' found for <project> in " + XmlUtils.toString(spotBugsTestEvent));
+                    continue;
+                }
+
+                xmlOutputDirectory = xmlOutputDirectory.replace("${basedir}", baseDir);
+            }
+
+            xmlOutputDirectory = XmlUtils.getPathInWorkspace(xmlOutputDirectory, workspace);
+
+            String spotBugsResultsFile = xmlOutputDirectory + "/spotbugsXml.xml";
```

Comment:
Since this only supports SpotBugs so far, maybe better to reflect that in the PR title.